### ISSUE TITLE
Problem when printing drivers timetable (Buchfahrpläne)

### ISSUE THREAD

**Issue by JTe-git** [FPLedit/FPLedit#32]
There is a problem when trying to print a pdf file of the drivers timetable (Buchfahrpläne) when there are the wavy lines for gradients (Wellenlinien).

The reason is that many of the PDF-converters do not handle well the embedded data:image/svg+xml;base64 format. If I change the imnage to data:image/png;base64 format everything works fine.

**Comment by ManuelHu** [comment]
Thanks for the report. First of all, what PDF converter are you using? I usually tested with the one included in Firefox and this generally should work decently, if I remember right.

FPLedit output is generally just normal HTML, and is expected to be used by software that is supporting a reasonable support of modern web tech.

About converting from SVG to PNG: I am not in favor of doing this, as this would siginificantly hinder users from adapting the templates to their own necessities. Nevertheless, I could investigate embedding the SVG data differently, as base64-encoded SVG is also not very user friendly.

**Comment by JTe-git** [comment]
I was testing with Firefox 110 (internal pdf), Safari 16.3, and Preview (using through Firefox) on macOS 12.6.3. All the three had problem with SVG. There is *no* problem when rendering on the screen, but when you try to print; the printed page or pdf created is an empty page. If the Wellenlinien is removed everything works fine. When I converted the SVG to PNG to test printing (pdf conversion) works fine.

**Comment by ManuelHu** [comment]
Thanks for the information.

I have also tested this in Firefox 112, by using the internal PDF printer available as "Save to PDF" in the print dialog. The generated PDFs view just fine in other PDF viewers. But note that I don't have any possibility to test on macOS currently.

I have added an option to output the images as PNG in 53734e00a745f26ce9281a034b7a6d25983868d8 and bd2fa0a5b3bb4f468b7e094bb8f91e958567993f. The option in the user-interface is located in the dialog "Fahrplandarstellung" and is only visible if the "Expert Mode" has been enabled.

This feature will be contained in the upcoming release of FPLedit 2.5.0, for which I released a beta version today. (note again, that the macOS version is essentially untested at the moment...)
